sqoop-tool/src/test/java/org/springframework/cloud/task/sqoop/tool/SqoopToolTaskApplicationTests.java

```diff
@@ -0,0 +1,148 @@
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.IntegrationTest;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.task.sqoop.common.SqoopToolDatabaseConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceUtils;
+import org.springframework.jdbc.datasource.SingleConnectionDataSource;
+import org.springframework.jdbc.datasource.init.DatabasePopulator;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * Verifies that the Sqoop Tool Task Application runs without throwing exception.
+ *
+ * NOTE: This test is using the local file system and running mapreduce in local mode.
+ *
+ * NOTE: We also start an HSQLDB database server with a data path of 'target/db'
+ *
+ * @author Thomas Risberg
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes={SqoopToolDatabaseConfiguration.class,
+		SqoopToolTaskApplicationTests.TestConfig.class, SqoopToolTaskApplication.class})
+@IntegrationTest
+public abstract class SqoopToolTaskApplicationTests {
+
+	private static String testDir;
+
+	static {
+		try {
+			testDir = Files.createTempDirectory("sqoop-test-").toFile().getAbsolutePath();
+			FileUtils.deleteDirectory(new File(testDir));
+		} catch (IOException e) {
+			testDir = System.getProperty("${java.io.tmpdir}") + "/sqoop-test-" + UUID.randomUUID();
+		}
+		System.setProperty("sqoop.test.dir", testDir);
+	}
+
+	@IntegrationTest({"spring.hadoop.fsUri=file:///",
+			"spring.hadoop.config.mapreduce.framework.name=local",
+			"connect=jdbc:hsqldb:hsql://localhost:${db.server.port}/test",
+			"username=sa",
+			"command=import",
+			"tool-args=--table FOO --target-dir ${sqoop.test.dir} -m 1 --outdir target/java"})
+	public static class SqoopToolImportTests extends SqoopToolTaskApplicationTests {
+
+		@Test
+		public void testImportTool() throws Exception {
+			File testOutput = new File(testDir);
+			assertTrue(testOutput.exists());
+			File[] files = testOutput.listFiles(new FilenameFilter() {
+				@Override
+				public boolean accept(File dir, String name) {
+					if (name.startsWith("part")) {
+						return true;
+					}
+					return false;
+				}
+			});
+			assertTrue("At least one data file should have been created", files.length > 0);
+		}
+	}
+
+	@IntegrationTest({"spring.hadoop.fsUri=file:///",
+			"spring.hadoop.config.mapreduce.framework.name=local",
+			"connect=jdbc:hsqldb:hsql://localhost:${db.server.port}/test",
+			"username=sa",
+			"command=export",
+			"tool-args=--table BAR --export-dir data --outdir target/java"})
```

Comment:
right, are those success and part files coming from here? maybe export to `target/data` instead of `data`?